components/private/Color/tests/unit/MaterialColorTests.m

```diff
@@ -77,4 +79,112 @@ - (void)testDynamicColorWhenUserInterfaceStyleIsLightForPreiOS13 {
+                                           green:(CGFloat)0.5
+                                            blue:(CGFloat)0.5
+                                           alpha:1];
+  UIColor *resultColor = [UIColor blendColor:blendColor withBackgroundColor:backgroundColor];
+  XCTAssertEqualObjects(resultColor, expectedColor);
+}
+
+- (void)testColorMergeFor60GrayOpacityOnWhite {
+  UIColor *backgroundColor = [UIColor whiteColor];
+  UIColor *blendColor = [UIColor colorWithRed:(CGFloat)0.9
+                                        green:(CGFloat)0.9
+                                         blue:(CGFloat)0.9
+                                        alpha:(CGFloat)0.6];
+  UIColor *resultColor = [UIColor blendColor:blendColor withBackgroundColor:backgroundColor];
+  UIColor *expectedColor = [UIColor colorWithRed:(CGFloat)0.94000000000000006
+                                           green:(CGFloat)0.94000000000000006
+                                            blue:(CGFloat)0.94000000000000006
+                                           alpha:(CGFloat)1];
+
+  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resultColor
+                                                    secondColor:expectedColor]);
+}
+
+- (void)testColorMergeFor50OpacityWhiteOnBlack {
+  UIColor *backgroundColor = [UIColor blackColor];
+  UIColor *blendColor = [UIColor colorWithRed:1 green:1 blue:1 alpha:(CGFloat)0.5];
+  UIColor *expectedColor = [UIColor colorWithRed:(CGFloat)0.5
+                                           green:(CGFloat)0.5
+                                            blue:(CGFloat)0.5
+                                           alpha:1];
+  UIColor *resultColor = [UIColor blendColor:blendColor withBackgroundColor:backgroundColor];
+  XCTAssertEqualObjects(resultColor, expectedColor);
+}
+
+- (void)testBasicColorMergeTest {
+  UIColor *backgroundColor = [UIColor colorWithRed:(CGFloat)0.4
+                                             green:(CGFloat)0.6
+                                              blue:(CGFloat)0.9
+                                             alpha:(CGFloat)0.8];
+  UIColor *blendColor = [UIColor colorWithRed:(CGFloat)0.1
+                                        green:(CGFloat)0.8
+                                         blue:(CGFloat)0.8
+                                        alpha:(CGFloat)0.2];
+  UIColor *expectedColor = [UIColor colorWithRed:(CGFloat)0.32857142857142863
+                                           green:(CGFloat)0.64761904761904765
+                                            blue:(CGFloat)0.87619047619047618
+                                           alpha:(CGFloat)0.84000000000000008];
+  UIColor *resultColor = [UIColor blendColor:blendColor withBackgroundColor:backgroundColor];
+  XCTAssertTrue([self compareColorsWithFloatPrecisionFirstColor:resultColor
+                                                    secondColor:expectedColor]);
+}
+
+- (void)testHBSColorMergeTest {
```

Comment:
Please also add tests for the P3 color space.

https://developer.apple.com/documentation/uikit/uicolor/1648568-init

Reply:
Addressed in 5b40c949c97f1940aa6bc79f4118b5a019fe3e3c